src/common/ram/ram.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ram
+
+import (
+	"path"
+)
+
+const (
+	// EffectAllow allow effect
+	EffectAllow = Effect("allow")
+	// EffectDeny deny effect
+	EffectDeny = Effect("deny")
+)
+
+// Resource the type of resource
+type Resource string
+
+func (res Resource) String() string {
+	return string(res)
+}
+
+// Subresource returns subresource
+func (res Resource) Subresource(resources ...Resource) Resource {
+	els := []string{res.String()}
```

Comment:
Can we change `els` with another meaningful term?

Reply:
els changed to elements